src/lib.rs

```diff
@@ -981,19 +982,15 @@ impl LineEventHandle {
 
     /// Helper function which returns the line event if a complete event was read, Ok(None) if not
     /// enough data was read or the error returned by `read()`.
-    ///
-    /// This function allows access to the raw `nix::Error` as required, for example, to theck
-    /// whether read() returned -EAGAIN.
-    pub(crate) fn read_event(&self) -> std::result::Result<Option<LineEvent>, nix::Error> {
+    pub(crate) fn read_event(&mut self) -> std::io::Result<Option<LineEvent>> {
         let mut data: ffi::gpioevent_data = unsafe { mem::zeroed() };
         let mut data_as_buf = unsafe {
             slice::from_raw_parts_mut(
                 &mut data as *mut ffi::gpioevent_data as *mut u8,
                 mem::size_of::<ffi::gpioevent_data>(),
             )
         };
-        let bytes_read = nix::unistd::read(self.file.as_raw_fd(), &mut data_as_buf)?;
-
+        let bytes_read = self.file.read(&mut data_as_buf)?;
         if bytes_read != mem::size_of::<ffi::gpioevent_data>() {
             Ok(None)
```

Comment:
again not really related to this PR (sorry, it's just caught my attention)... do you think there is any case in which a partial read should occur during operation? 

i wonder whether this line is hiding dropping some data / a partial cause of https://github.com/rust-embedded/gpio-cdev/issues/51, and whether this should instead be something like:

```
if bytes_read == 0 {
  return Ok(None);
else if bytes_read < mem::sizeof::... {
  // Error? Flush (drop events but recover)?
} ...
```

Reply:
Hm, possibly. Do you know if a gpio char device can return partial packets? Couldn't find anything about that.

I couldn't repeat that `select!` error locally either, tested with master branch. Not running on a raspberry though.

Reply:
With the way that the file works in the kernel for a character device, I wouldn't ever expect a read call to return a partial structure.  I'll have to consider the linked issue but I don't think that branch would ever be hit in practice.